### PR TITLE
ci: introducing semantic versioning to mason.nvim

### PR DESCRIPTION
**Pull request by jay-babu**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        - cbfmt
+        - selene
+        - stylua
+        - tests
+      runs-on: ubuntu-latest
+      steps:
+        - uses: google-github-actions/release-please-action@v3
+          id: release
+          with:
+            token: ${{ secrets.RELEASE_TOKEN }}
```
Comment (williamboman) [inline .github/workflows/ci.yml:94]: Can we use the default secret (`GITHUB_TOKEN`) instead (i.e. omit this entirely)?

**Review by williamboman — COMMENTED** [review]
Sweet! Any reason to merge all workflows into one? I think it'd be neat to keep them separated. I think it's fine if the release job doesn't explicitly depend on the other jobs, as the `main` branch is protected anyway.

It also seems like the release job needs some extra permissions, so it'd be they were only applied for that job (it's probably possible to configure this on a per-job basis though):

```yaml
permissions:
  contents: write
  pull-requests: write
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jay-babu) [reply .github/workflows/ci.yml:94]: yes, sorry. bad copy paste

**Comment by jay-babu** [comment]
> Sweet! Any reason to merge all workflows into one? I think it'd be neat to keep them separated. I think it's fine if the release job doesn't explicitly depend on the other jobs, as the `main` branch is protected anyway.
> 
> It also seems like the release job needs some extra permissions, so it'd be they were only applied for that job (it's probably possible to configure this on a per-job basis though):
> 
> ```yaml
> permissions:
>   contents: write
>   pull-requests: write
> ```

okay, if you are fine with not having the release job depend on the others, then they can be separate

**Comment by williamboman** [comment]
Sweet, let's give this a try

**Pull request merged**